Fix variable replacement in topic string

pierrepitiot opened the pull request:
In Wires publisher, rather than limiting the variable replacement to only "assetName", all the variables from a record should be available.

**Related Issue:** This PR fixes/closes #2448

pierrepitiot committed:
Fix variable replacement in topic string
Rather than limiting the variable replacement to only "assetName", all
the variables from a record should be available.

MMaiero commented on kura/org.eclipse.kura.wire.component.provider/src/main/java/org/eclipse/kura/internal/wire/publisher/CloudPublisher.java:
```diff
@@ -295,8 +296,10 @@ private void publish(final List<WireRecord> wireRecords) {
-        if (wireRecordProps.containsKey(ASSET_NAME_PROPERTY_KEY)) {
-            properties.put(ASSET_NAME_PROPERTY_KEY, wireRecordProps.get(ASSET_NAME_PROPERTY_KEY).getValue());
+        List<String> l = new ArrayList<>(wireRecordProps.keySet());
+        for(String s:l) {
```
please apply formatting

MMaiero reviewed (commented):
Could you have a look if it is possible to add unit/integration tests to prove that the substitution works?

pierrepitiot committed:
Apply formatter

pierrepitiot committed:
remove debug line

pierrepitiot commented:
@MMaiero I am afraid I don't know how to do that ...

MMaiero commented:
Have a look here: https://github.com/eclipse/kura/tree/develop/kura/test/org.eclipse.kura.wire.component.provider.test

pierrepitiot commented:
@MMaiero , sorry but I don't find how I can test the topic in this test. the CloudPublisher that is taking care of that is a mock in this test and I didn't find how to replace it with a real CloudPublisher...

MMaiero commented:
Can't you just do a unit test verifying that the substitution is performed correctly by your code?

pierrepitiot committed:
Adding a test for variable substitution in topic string

(not sure this is done properly...)

MMaiero reviewed (commented):
@pierrepitiot Why didn't you add the test to the wire.component.provider.test project?

pierrepitiot commented:
Because I made a lot of different tries and it was simpler to use a new project but I can merge it in wire.component.provider.test if you think it's better

MMaiero commented:
Yes please, it would match with what we already have

pierrepitiot committed:
Remove specific topic test and include it in
wire.component.provider.test

By the way, the initial tests of this project are failing
